crates/ide-db/src/rename.rs

```diff
@@ -71,12 +71,25 @@ impl Definition {
         sema: &Semantics<'_, RootDatabase>,
         new_name: &str,
     ) -> Result<SourceChange> {
+        if let Some(krate) = self.krate(sema.db) {
```

Comment:
Can you add a comment that `self.krate()` is `None` when `self` is either a built-in attribute, built-in type, or tool module, and the former two are handled below in a match?

I wonder if we should bail out here as well when the def is a tool module.